chalk-ir/src/interner.rs

```diff
@@ -123,6 +127,23 @@ pub trait Interner: Debug + Copy + Eq + Ord + Hash {
     /// and can be converted back to its underlying data via `quantified_where_clauses_data`.
     type InternedQuantifiedWhereClauses: Debug + Clone + Eq + Hash;
 
+    /// "Interned" representation of a list of parameter kind.  
+    /// In normal user code, `Self::InternedParameterKinds` is not referenced.
+    /// Instead, we refer to `ParameterKinds<Self>`, which wraps this type.
+    ///
+    /// An `InternedParameterKinds` is created by `intern_parameter_kinds`
+    /// and can be converted back to its underlying data via `parameter_kinds_data`.
+    type InternedParameterKinds: Debug + Clone + Eq + Hash;
+
+    /// "Interned" representation of a list of parameter kind with universe index.  
+    /// In normal user code, `Self::InternedParameterKindsWithUniverseIndex` is not referenced.
+    /// Instead, we refer to `ParameterKindsWithUniverseIndex<Self>`, which wraps this type.
+    ///
+    /// An `InternedParameterKindsWithUniverseIndex` is created by
+    /// `intern_parameter_kinds_with_universe_index` and can be converted back
+    /// to its underlying data via `parameter_kinds_with_universe_index_data`.
+    type InternedParameterKindsWithUniverseIndex: Debug + Clone + Eq + Hash;
```

Comment:
Can we rename this to `InternedCanonicalVarKinds`?

Reply:
(And rename the wrapper to `CanonicalVarKinds` and the like)

Reply:
In our discussion from Zulip we decided we should do this before landing

Reply:
Done.